### PR TITLE
feat(bigtable): adds readmodifywriterow to proxy

### PR DESCRIPTION
**Pull request by telpirion**

### CODE REVIEW

File: bigtable/internal/testproxy/proxy.go [hunk 1]
```diff
@@ -801,7 +801,57 @@ func (s *goTestProxyServer) SampleRowKeys(ctx context.Context, req *pb.SampleRow
...
+		return nil, stat.Error(codes.InvalidArgument, "request to CheckAndMutateRow() is missing inner request")
+	}
+
+	rpb := rrq.Rules
+	rmw := bigtable.NewReadModifyWrite()
+
+	for _, rp := range rpb {
+		switch r := rp.Rule; r.(type) {
+		case *btpb.ReadModifyWriteRule_AppendValue:
+			av := r.(*btpb.ReadModifyWriteRule_AppendValue)
```
Comment (enocom) [inline bigtable/internal/testproxy/proxy.go:826]: Let's updated the switch statement to avoid this duplicate type casting.

**Review by enocom — CHANGES_REQUESTED** [review]

File: bigtable/internal/testproxy/proxy.go [hunk 2]
```diff
@@ -853,7 +853,57 @@ func (s *goTestProxyServer) SampleRowKeys(ctx context.Context, req *pb.SampleRow
...
+		return nil, stat.Error(codes.InvalidArgument, "request to CheckAndMutateRow() is missing inner request")
+	}
+
+	rpb := rrq.Rules
+	rmw := bigtable.NewReadModifyWrite()
+
+	for _, rp := range rpb {
+		switch r := rp.Rule.(type) {
+		case *btpb.ReadModifyWriteRule_AppendValue:
+			av := r
```
Comment (enocom) [inline bigtable/internal/testproxy/proxy.go:878]: nit: These are superfluous.

**Review by enocom — APPROVED** [review]

File: bigtable/internal/testproxy/proxy.go (hunk 1 above)
Comment (telpirion) [reply bigtable/internal/testproxy/proxy.go]: Done.

File: bigtable/internal/testproxy/proxy.go (hunk 2 above)
Comment (telpirion) [reply bigtable/internal/testproxy/proxy.go:878]: Ack.

**Pull request merged**